cli/command/stack/deploy_composefile.go

```diff
@@ -320,6 +322,13 @@ func deployServices(
 			if sendAuth {
 				updateOpts.EncodedRegistryAuth = encodedAuth
 			}
+
+			if image != service.Spec.TaskTemplate.ContainerSpec.Image {
```

Comment:
@aaronlehmann you're right, that would be a problem.

For `service update`, it's pretty simple to judge because we have a `--image` flag, but I couldn't find a way of deciding whether an image was updated in an updated stack file. Is there a way of discovering this at the CLI level? cc @dnephin